src/utils/send-request.js

```diff
@@ -13,9 +13,21 @@ const log = require('debug')('ipfs-http-client:request')
 
 // -- Internal
 
-function parseError (res, cb) {
+function parseError (res, cb, isJson = true) {
   const error = new Error(`Server responded with ${res.statusCode}`)
 
+  if (!isJson) {
+    return streamToValue(res, (err, data) => {
+      // the `err` here refers to errors in stream processing, which
+      // we ignore here, since we already have a valid `error` response
+      // from the server above that we have to report to the caller.
+      if (data) {
```

Comment:
So we don't get a blank error message!

```suggestion
      if (data && data.length) {
```

Reply:
Yes - had that nagging feeling `data` should be non-empty (in addition to being non-null), but for some reason was hesitant to assume the `.length` property exists on the `data` object in all valid cases. The hesitation came from this [Line 19](https://github.com/ipfs/js-ipfs-http-client/blob/master/src/utils/stream-to-json-value.js#L19) where data is being checked if it is a buffer: `if (Buffer.isBuffer(data))`

But now that I think about it, you are right. Even for empty strings the `.length` is a valid value in Javascript (and confirmed it in Chrome console). Committing your suggestion. Hope `buffer` and `string` are the only two cases the `data` can be of type.